src/hooks/useConfig.ts

```diff
@@ -0,0 +1,50 @@
+import {useState} from 'react';
+
+import {Config, MidiInputConfig} from '../types/config-types';
+import {mockMidiInputConfig} from '../mockfiles/MockMidiDevice';
+
+export type UseConfigHookValue = {
+  config: Config;
+
+  setConfig: (config: Config) => void;
+  updateConfigForInputDevice: (inputConfig: MidiInputConfig) => void;
+}
+
+const initialConfig: Config = {
+  midi: {
+    inputs: [
+      mockMidiInputConfig,
+    ],
+    outputs: [],
+  }
+}
+
+export const useConfig = (): UseConfigHookValue => {
+  const [config, setConfig] = useState<Config>(initialConfig);
```

Comment:
The `useState` and `useMidi` hooks should really be using Context instead of `useState`. That way multiple React components can use the hook and reference the same instance of WebMidi etc.